### ISSUE TITLE
Expose richer price info in search results

### ISSUE THREAD

**Issue by michaelbromley** [vendure-ecommerce/vendure#82]
Currently the search query returns a `price` property, which represents the net (tax-exclusive) price of that ProductVariant. We also need to expose the gross (tax-inclusive) price, since many sites will want to display this price by default.

Also, then `groupByProduct` is set to `true`, the returned `price` property is not meaningful for Products with more than one variant. A more useful property would be `minPrice` and `maxPrice`, which represent the lowest and highest prices in the range. 

When not grouped by product, the `minPrice`, `maxPrice` and `price` properties would all be equal.

To summarize, the following properties should be added to the `SearchResult` type:

* `minPrice`
* `maxPrice`
* `priceWithTax`
* `minPriceWithTax`
* `maxPriceWithTax`

Or, maybe instead of using an `Int` type, we create a `PriceRange` subtype:

```sdl
type PriceRange {
  value: Int!
  min: Int!
  max: Int!
}
```

**Comment by michaelbromley** [comment]
I decided to create a union type for the price and priceWithTax properties. This strikes a balance between being verbose, and providing context-relevant results.
